src/backend/src/utils/projects.utils.ts

```diff
@@ -17,6 +17,20 @@ import { riskQueryArgs, riskTransformer } from './risks.utils';
 import { buildChangeDetail } from '../utils/utils';
 import { calculateWorkPackageProgress } from './work-packages.utils';
 
+/**
+ * calculate the project's status based on its workpacakges' status
+ * @param proj a project where its status will be updated based on its workpackges' status
```

Comment:
this function isn't actually doing the updating though, it's just calculating what it should be based off of some cases